apps/emqx_bridge_kafka/src/emqx_bridge_kafka_impl_producer.erl

```diff
@@ -337,6 +366,28 @@ render_timestamp(Template, Message) ->
             erlang:system_time(millisecond)
     end.
 
+do_send_msg(sync, KafkaMessage, Producers, SyncTimeout) ->
+    try
+        {_Partition, _Offset} = wolff:send_sync(Producers, [KafkaMessage], SyncTimeout),
+        ok
+    catch
+        error:{producer_down, _} = Reason ->
```

Comment:
if this can happen, we are in a more serious situation, we should fix wolff instead.

if many invalid messages are sent around the same time, it will cause all producers to crash,
then bring down the supervisor, then its supervisor etc.

Reply:
or add a filter here to drop invalid headers.

Reply:
I didn't get your point. The invalid headers are being handled on the `render_message` procedure, which will throw a bad header exception that is being caught by `on_query`/`on_query_async`, so these messages do not reach the wolff lib.